dbt_semantic_interfaces/transformations/rule_set.py

```diff
@@ -0,0 +1,31 @@
+import logging
+from abc import abstractmethod
+from typing import Protocol, Sequence
+
+from dbt_semantic_interfaces.transformations.transform_rule import (
+    SemanticManifestTransformRule,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class SemanticManifestTransformRuleSet(Protocol):
+    """Groups rules that should be run for a SemanticManifest."""
+
+    @property
+    @abstractmethod
+    def primary_rules(self) -> Sequence[SemanticManifestTransformRule]:
+        """TODO: Define what primary means."""
+        pass
+
+    @property
+    @abstractmethod
+    def secondary_rules(self) -> Sequence[SemanticManifestTransformRule]:
+        """TODO: Define what secondary means."""
+        pass
+
+    @property
+    @abstractmethod
+    def all_rules(self) -> Sequence[Sequence[SemanticManifestTransformRule]]:
+        """TODO: Why a nested sequence?"""
```

Comment:
Oh yeah, this is a legacy holdover from when we used to do transform -> validate -> transform -> validate, except we never actually did that, it's just the interfaces all pretended like we did. 

I don't think we ever break up the transform rules so we can simply unnest and do primary + secondary here.